ports/apr/portfile.cmake

```
@@ -10,31 +10,38 @@ vcpkg_download_distfile(ARCHIVE
 vcpkg_extract_source_archive(SOURCE_PATH
     ARCHIVE "${ARCHIVE}"
     PATCHES
-        fix-configcmake.patch
+        apr-r1917051.patch
+        apr-r1917065.patch
+        apr-r1917283.patch
+        apr-r1917612.patch
         unglue.patch
 )
 
 if (VCPKG_TARGET_IS_WINDOWS)
     vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
-        FEATURES
-            private-headers APR_INSTALL_PRIVATE_H
+    FEATURES
+        private-headers INSTALL_PRIVATE_H
     )
 
+    string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" APR_BUILD_STATIC)
+    string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" APR_BUILD_SHARED)
+
     vcpkg_cmake_configure(
-        SOURCE_PATH "${SOURCE_PATH}"
-        OPTIONS
-            -DINSTALL_PDB=OFF
-            -DMIN_WINDOWS_VER=Windows7
-            -DAPR_HAVE_IPV6=ON
-            ${FEATURE_OPTIONS}
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        -DAPR_BUILD_STATIC=${APR_BUILD_STATIC}
+        -DAPR_BUILD_SHARED=${APR_BUILD_SHARED}
+        -DAPR_BUILD_TESTAPR=OFF
+        -DINSTALL_PDB=OFF
+        -DAPR_INSTALL_PRIVATE_H=${INSTALL_PRIVATE_H}
```

Indent and revert `${FEATURE_OPTIONS}`
```suggestion
        SOURCE_PATH "${SOURCE_PATH}"
        OPTIONS
            -DAPR_BUILD_STATIC=${APR_BUILD_STATIC}
            -DAPR_BUILD_SHARED=${APR_BUILD_SHARED}
            -DAPR_BUILD_TESTAPR=OFF
            -DINSTALL_PDB=OFF
            ${FEATURE_OPTIONS}
```

---

I reverted accidentally changed indentation in https://github.com/microsoft/vcpkg/pull/38674/commits/faa5eb2e887cd0488e7f951099fc009b83baf1c9

About `${FEATURE_OPTIONS}` I disagree, because the feature variable name is `INSTALL_PRIVATE_H` while the cmake command `APR_INSTALL_PRIVATE_H`, so it would not work.

Additionally, I think that it is better to pass the options explicitly.

---

`vcpkg_check_feautures` exist only to provide feature options to cmake. The original `APR_INSTALL_PRIVATE_H` / `FEATURE_OPTIONS` pattern is "canonical vcpkg". Please restore it.